openedx/features/enterprise_support/tests/test_api.py

```diff
@@ -168,15 +170,28 @@ def test_enterprise_consent_api_client_with_user(self, mock_jwt_builder):
     def test_consent_needed_for_course(self):
         user = UserFactory(username='janedoe')
         request = mock.MagicMock(session={}, user=user)
+        ec_uuid = 'cf246b88-d5f6-4908-a522-fc307e0b0c59'
+        course_id = 'fake-course'
         self.mock_enterprise_learner_api()
-        self.mock_consent_missing(user.username, 'fake-course', 'cf246b88-d5f6-4908-a522-fc307e0b0c59')
-        self.assertTrue(consent_needed_for_course(request, user, 'fake-course'))
-        self.mock_consent_get(user.username, 'fake-course', 'cf246b88-d5f6-4908-a522-fc307e0b0c59')
-        self.assertFalse(consent_needed_for_course(request, user, 'fake-course'))
-        # Test that the result is cached when false (remove the HTTP mock so if the result
-        # isn't cached, we'll fail spectacularly.)
-        httpretty.reset()
-        self.assertFalse(consent_needed_for_course(request, user, 'fake-course'))
+
+        # test not required consent for example non enterprise customer
+        self.mock_consent_not_required(user.username, course_id, ec_uuid)
+        self.assertFalse(consent_needed_for_course(request, user, course_id))
+
+        # test required and missing consent for example now he becomes a enterprise customer
+        self.mock_consent_missing(user.username, course_id, ec_uuid)
+        # still result should be False as it has been stored in cache "Not to show consent", so it will confirm that
+        # cache is working fine
+        self.assertFalse(consent_needed_for_course(request, user, course_id))
+        # Removing cache
+        consent_cache_key = get_data_consent_share_cache_key(user.id, course_id)
+        TieredCache.delete_all_tiers(consent_cache_key)
```

Comment:
Instead of manually deleting cache here you should update `EnterpriseCourseEnrollment` record to emit relevant signal.

Reply:
As we are using mocked APIs so we are not actually creating `EnterpriseCourseEnrollment` object.
To make it better I have shifted this logic into a common method  and called it from `update_data_consent_share_cache` **signal** and `test_consent_needed_for_course` **test**